### ISSUE TITLE
stan_lm silently ignores autoscale argument if prior_intercept is normal

### ISSUE THREAD

**Issue by jgabry** [stan-dev/rstanarm#353]
#### Summary:

`stan_lm()` (really `stan_biglm.fit` and everything it's used for) seems to silently ignore the `autoscale` argument to `normal()` when used for `prior_intercept`. @bgoodri Is ignoring `autoscale` a bug or intended? I guess regardless of whether it's a bug or not, I don't think it should be silent. 

#### Reproducible Steps:

```r
# fit same model with same seed, only difference is autoscale argument
fit <- stan_lm(mpg ~ wt + qsec + am, data = mtcars, 
               prior = R2(0.75), prior_intercept = normal(0, 10, autoscale=TRUE),
               chains = 1, iter = 500, seed = 12345)
fit2 <- update(fit1, prior_intercept = normal(0, 10, autoscale=FALSE))

# draws are identical
identical(as.matrix(fit1), as.matrix(fit2)) 
[1] TRUE
```

#### RStanARM Version:
2.18.9

#### R Version:
3.5.2

#### Operating System:
OS X 10.12.6

**Comment by bgoodri** [comment]
It isn't really supposed to auto-scale the priors like it does with
stan_glm because stan_lm is estimating a marginal scale parameter for the
outcome rather than cheating. But that does not really work with
prior_summary because the scaling is not fixed. I think if someone does
prior_intercept = normal(autoscale = FALSE), then we should not be doing
any scaling, but we would need to pass a flag to lm.stan saying whether the
user did autoscale = FALSE or autoscale = TRUE.

**Comment by jgabry** [comment]
Yeah it would be good to change that in lm.stan

**Comment by bgoodri** [comment]
But I don't think we have a good way to represent that in prior_summary()

**Comment by jgabry** [comment]
I agree that we don't have a good way to include it in prior_summary. Maybe we could have the prior_summary output indicate to look at some part of the doc that explains it. That would at least be better than the status quo I think.
